packages/auth/src/mfa/assertions/totp.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TotpSecret } from '../../platform_browser/mfa/assertions/totp';
+import { TotpMultiFactorAssertion } from '../../model/public_types';
+import { FactorId } from '../../model/enum_maps';
+import { MultiFactorSession } from '../../model/public_types';
+/**
+ * Provider for generating a {@link TotpMultiFactorAssertion}.
+ *
+ * @public
+ */
+export class TotpMultiFactorGenerator {
+  /**
+   * Provides a {@link TotpMultiFactorAssertion} to confirm ownership of the totp(Time-based One Time Password) second factor.
+   * This assertion is used to complete enrollment in TOTP second factor.
+   *
+   * @param secret {@link TotpSecret}.
+   * @param oneTimePassword One-time password from TOTP App.
+   * @returns A {@link TotpMultiFactorAssertion} which can be used with
+   * {@link MultiFactorUser.enroll}.
+   */
+  static assertionForEnrollment(
+    secret: TotpSecret,
```

Comment:
nit: can you mark all these as unused, by using names like _secret, _oneTimePassword? Thanks!